Tribler/Test/Core/CreditMining/test_creditmining_sys.py

```diff
@@ -498,29 +478,19 @@ def test_chn_max_torrents(self):
         self.boosting_manager.add_source(dispersy_cid)
         chn_obj = self.boosting_manager.get_source_object(dispersy_cid)
         chn_obj.max_torrents = 2
-        chn_obj.torrent_mgr.load_torrent = lambda dummy_1, dummy_2: None
-
-        def _load(torrent, callback=None):
-            if not isinstance(torrent, CollectedTorrent):
-                torrent_id = 0
-                if torrent.torrent_id <= 0:
-                    torrent_id = self.session.lm.torrent_db.getTorrentID(torrent.infohash)
-                if torrent_id:
-                    torrent.update_torrent_id(torrent_id)
-
-                infohash_str = binascii.hexlify(torrent.infohash)
-                torrent = CollectedTorrent(torrent, self.tdef if infohash_str.startswith("fc") else pioneer_tdef)
-            if callback is not None:
-                callback(torrent)
-            else:
-                return torrent
+        chn_obj._load_torrent = lambda _: defer.Deferred()
+
+        def _load(infohash):
+            defer_ret = defer.Deferred()
+            defer_ret.callback(self.tdef if binascii.hexlify(infohash).startswith("fc") else pioneer_tdef)
+            return defer_ret
```

Comment:
Same comment as above (I acknowledge this `_load` method is slightly different so it might be a bit harder to move this method to a higher level)).